public/js/notification-checkbox.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright 2015-2016, Google, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* eslint-env browser */
+
+import Messaging from './messaging';
+
+export default class NotificationCheckbox {
+  init(firebaseMsgSenderId, checkbox, topic) {
+    if (!checkbox) {
+      return;
+    }
+    this.messaging = new Messaging();
```

Comment:
This feels like it doesn't belong inside this class, but should be passed in as a dependency via the constructor. Multiple classes might need to access messaging...